### PR TITLE
Feat/front

### PR DESCRIPTION
**Pull request by valter-junior**
Review

### CODE REVIEW

File: front/src/App.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      
+
+      </Container>
+      
+    </div>
+  )
+}
+
+
+export default App;
```
Comment (jfmrm) [inline front/src/App.js]: add empty line on the file end

File: front/src/App.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+
+
+
+function App () {
+  return (
+    <div>
+      <Container>
+       
+
```
Comment (jfmrm) [inline front/src/App.js]: remove this blank space

File: front/src/App.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  return (
+    <div>
+      <Container>
+       
+      
+          <Switch>
+            <ProtectedRoute exact path="/login" component={Home}>
+              <Login />
+            </ProtectedRoute>
+
```
Comment (jfmrm) [inline front/src/App.js]: here too

File: front/src/App.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+              
+            <Route path="/sign-in" component={SignIn}>
+              <SignIn />
+            </Route>
+            <Route path="/">
+              <Home />
+            </Route>
+          </Switch>
+      
+
```
Comment (jfmrm) [inline front/src/App.js]: here too

File: front/src/App.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+      </Container>
+
```
Comment (jfmrm) [inline front/src/App.js]: here too

File: front/src/auth.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+        this.authenticated = false;
+        cb();
+    }
+    
+    isAuthenticated() {
+        return this.authenticated;
+    }
+}
+
+export default new Auth();
```
Comment (jfmrm) [inline front/src/auth.js]: add empty line

File: front/src/home.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+export const Home = (props) => {
+    const history = useHistory();
+    return (
+        <div>
+        <h1>Task</h1>
+        <Button 
+            onClick={() => {
+                auth.login(() => {
+                    history.push("/login")
+
```
Comment (jfmrm) [inline front/src/home.js]: remove empty line

File: front/src/home.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+        <h1>Task</h1>
+        <Button 
+            onClick={() => {
+                auth.login(() => {
+                    history.push("/login")
+                    
+                });
+            }}>Login</Button>
+        <Button component={Link} to="/sign-in">Sign In</Button>
+    </div>
```
Comment (jfmrm) [inline front/src/home.js]: needs to be indented

File: front/src/index.js [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+  <BrowserRouter>
+    <App />
+  </BrowserRouter>,
+  document.getElementById('root')
+);
+
+// If you want to start measuring performance in your app, pass a function
+// to log results (for example: reportWebVitals(console.log))
+// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
+reportWebVitals();
```
Comment (jfmrm) [inline front/src/index.js]: add empty line

File: front/src/login/login.js [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+                <input type="password" {...register("password")} />
+            </label>
+            
+            <input type="submit" value="submit" />
+
+        </form>
+    );
+}
+
+export default Login;
```
Comment (jfmrm) [inline front/src/login/login.js]: add empty line

File: front/src/login/login.js [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+                Email:
+                <input {...register("login")} />
+            </label>
+            <label>
+                Password:
+                <input type="password" {...register("password")} />
+            </label>
+            
+            <input type="submit" value="submit" />
+
```
Comment (jfmrm) [inline front/src/login/login.js]: remove blank spaces

File: front/src/login/sign.in.js [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+                Email:
+                <input {...register("login")} />
+            </label>
+            <label>
+                Password:
+                <input type="password" {...register("password")} />
+            </label>
+                
+            <input type="submit" value="submit" />
+
```
Comment (jfmrm) [inline front/src/login/sign.in.js]: remove empty lines

File: front/src/login/sign.in.js [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+        </form>
+    );
+}
+
+export default SignIn;
```
Comment (jfmrm) [inline front/src/login/sign.in.js]: add empty line

File: front/src/protected.route.js [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+                        }
+                    }
+                }/>
+
+            }
+                
+            }
+        }/>
+    )
+}
```
Comment (jfmrm) [inline front/src/protected.route.js]: add empty line

File: front/src/protected.route.js [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+                        state: {
+                            from: props.location
+                        }
+                    }
+                }/>
+
+            }
+                
+            }
+        }/>
```
Comment (jfmrm) [inline front/src/protected.route.js]: fix indentation

**Review by jfmrm — CHANGES_REQUESTED** [review]

**Comment by jfmrm** [comment]
Now its compiling, nice job!

**Comment by jfmrm** [comment]
one question: why have you used BrouserRouter on the index.js file?

**Comment by valter-junior** [comment]
> one question: why have you used BrouserRouter on the index.js file?

I'm not right about this but  I used this component in file index.js to render my function with route written in file App.js and I forget fix it

File: front/src/home.js [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+            <div>
+            <h1>Task</h1>
+            <Button 
+                onClick={() => {
+                    auth.login(() => {
+                        history.push("/login")       
+                    });
+                }}>Login</Button>
+            <Button component={Link} to="/sign-in">Sign In</Button>
+        </div>
```
Comment (jfmrm) [inline front/src/home.js]: still not indented right do it like this:
```suggestion
              <h1>Task</h1>
              <Button 
                  onClick={() => {
                      auth.login(() => {
                          history.push("/login")       
                      });
                  }}>Login</Button>
              <Button component={Link} to="/sign-in">Sign In</Button>
          </div>
```

File: front/src/protected.route.js [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+                    {
+                        pathname: "/",
+                        state: {
+                            from: props.location
+                        }
+                    }
+                }/>
+
+            }
+
```
Comment (jfmrm) [inline front/src/protected.route.js]:
```suggestion
              if(auth.isAuthenticated()) {
                  return <Component {...props}/>
              } else {
                  return (
                      <Redirect to={{
                          pathname: "/",
                          state: {
                              from: props.location
                         }}
                      }/>
                  )
             }   
```

File: front/src/App.js [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+              <Login />
+            </ProtectedRoute>
+            <Route path="/sign-in" component={SignIn}>
+              <SignIn />
+            </Route>
+            <Route path="/">
+              <Home />
+            </Route>
+          </Switch>
+      </Container>
```
Comment (jfmrm) [inline front/src/App.js]: where is the Browser Router?

**Review by jfmrm — CHANGES_REQUESTED** [review]

**Comment by valter-junior** [comment]
Changes done

File: front/src/home.js [hunk 19]
```diff
@@ -0,0 +1,25 @@
...
+               <h1>Task</h1>
+                 <Button 
+                    onClick={() => {
+                      auth.login(() => {
+                          history.push("/login")       
+                      });
+                    }}>Login</Button>
+                 <Button component={Link} to="/sign-in">Sign In</Button>
+             </div>
+    )
```
Comment (jfmrm) [inline front/src/home.js]:
```suggestion
    return (
       <div>
         <h1>Task</h1>
           <Button 
              onClick={() => {
                auth.login(() => {
                    history.push("/login")       
                });
              }}>Login</Button>
           <Button component={Link} to="/sign-in">Sign In</Button>
       </div>
    )
```

File: front/src/index.js [hunk 20]
```diff
@@ -0,0 +1,20 @@
...
+import { BrowserRouter } from 'react-router-dom';
+import App from './App';
+import reportWebVitals from './reportWebVitals';
+
+
+
+ReactDOM.render(
+  <BrowserRouter>
+    <App />
+  </BrowserRouter>,
```
Comment (jfmrm) [inline front/src/index.js]: should be React.StrictMode here.

BrowserRouter is already defined inside App

**Review by jfmrm — CHANGES_REQUESTED** [review]

**Comment by valter-junior** [comment]
Done

**Review by jfmrm — APPROVED** [review]

**Pull request merged**